### PR TITLE
Allow for reading imaginary vibrational frequencies in ORCA outputs.

### PR DESCRIPTION
**Pull request by dtelsing**
When reading an ORCA output file with an imaginary mode (which have an ***imaginary mode*** comment in the frequency line), the line and subsequent lines are skipped. This leads to an out-of-bounds access when reading normal modes, the IR spectrum or the Raman spectrum, as the indices read from the file are not verified.

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

File: avogadro/quantumio/orca.cpp [hunk 1]
```diff
@@ -397,6 +415,9 @@ void ORCAOutput::processLine(std::istream& in, GaussianSet* basis)
           // the first entry might be 5 or 6 because of removed rotations /
           // translations
           int index = Core::lexicalCast<int>(list[0]);
+          // invalid index
```
Comment (ghutchis) [inline avogadro/quantumio/orca.cpp:418]: As above.

File: avogadro/quantumio/orca.cpp [hunk 2]
```diff
@@ -332,13 +336,23 @@ void ORCAOutput::processLine(std::istream& in, GaussianSet* basis)
...
         list = Core::split(key, ' ');
         vector<int> modeIndex;
+        bool invalid_index = false;
         while (!key.empty()) {
           // first we get a set of column numbers
           // e.g. 1  2  3  4  5  6  7  8  9 10
           modeIndex.clear();
           for (unsigned int i = 0; i < list.size(); i++) {
-            modeIndex.push_back(Core::lexicalCast<int>(list[i]));
+            int index = Core::lexicalCast<int>(list[i]);
```
Comment (ghutchis) [inline avogadro/quantumio/orca.cpp:345]: Shouldn't we just declare (index) as `std::size_t` here?

File: avogadro/quantumio/orca.cpp [hunk 3]
```diff
@@ -374,6 +388,10 @@ void ORCAOutput::processLine(std::istream& in, GaussianSet* basis)
           // the first entry might be 5 or 6 because of removed rotations /
           // translations
           int index = Core::lexicalCast<int>(list[0]);
+          // invalid index
```
Comment (ghutchis) [inline avogadro/quantumio/orca.cpp:391]: Shouldn't we just declare (index) as `std::size_t` above? In that case, `index > 0` below can be dropped too.

**Comment by ghutchis** [comment]
Great, thanks! I obviously haven't dealt with many TS or imaginary modes and was largely porting the "Orca-enhanced Avogadro 1.2" code.

**Comment by ghutchis** [comment]
The CMake build failure on Ubuntu / GCC seems like a problem on that build node. I'll restart in an hour or two. (There are lots of new PR, so each is taking a while.)

**Comment by ghutchis** [comment]
If you can make the minor changes I mentioned, I'm happy to merge this ASAP. Thanks!

File: avogadro/quantumio/orca.cpp (hunk 2 above)
Comment (dtelsing) [reply avogadro/quantumio/orca.cpp]: Yes, I just kept it as it was before.

File: avogadro/quantumio/orca.cpp (hunk 3 above)
Comment (dtelsing) [reply avogadro/quantumio/orca.cpp:391]: Yes.

**Pull request merged**